crates/common/src/proofs.rs

```diff
@@ -67,7 +65,11 @@ impl SignableBLS for ConstraintsMessage {
         hasher.update(&self.slot.to_le_bytes());
         hasher.update((self.top as u8).to_le_bytes());
         for tx in self.transactions.iter() {
-            hasher.update(&keccak256(tx.to_vec()).as_slice());
+            // Convert the opaque bytes to a EIP-2718 envelope and obtain the tx hash.
+            // this is needed to handle type 3 transactions.
+            // FIXME: don't unwrap here and handle the error properly
+            let tx = PooledTransactionsElement::decode_enveloped(tx.to_vec().into()).unwrap();
+            hasher.update(tx.hash().as_slice());
```

Comment:
Can we either modify or track it in an issue? @merklefruit

Reply:
Tracked in #12. It would fail on decoding instead of here.